### ISSUE TITLE
The Julia language server cannot go up

### ISSUE THREAD

**Issue by roiholtzman** [gdkrmr/lsp-julia#46]
I have posted this [issue](https://github.com/emacs-lsp/lsp-mode/issues/2722) on the [emacs-lsp](https://github.com/emacs-lsp) project, but maybe it's related to lsp-julia.

This is the content of the issue:
The Julia language server, `lsp-julia` cannot start.

I use doom emacs on MacOS. I upgraded doom and the problem continues. 
A surprising thing is that initially everything worked great. Then I restarted emacs and the problem appeared. Using some help on discord I upgraded doom emacs, but still things are not working.

The error from `Julia-ls::stderr`:
```
ERROR: MethodError: no method matching LanguageServer.FoldingRangeClientCapabilities(::Nothing)
Closest candidates are:
  LanguageServer.FoldingRangeClientCapabilities(::Any, !Matched::Any, !Matched::Any) at /Users/roiholtzman/.julia/packages/LanguageServer/y1ebo/src/protocol/features.jl:217
  LanguageServer.FoldingRangeClientCapabilities(; dynamicRegistration, rangeLimit, lineFoldingOnly) at /Users/roiholtzman/.julia/packages/JSONRPC/xbAlZ/src/interface_def.jl:50
  LanguageServer.FoldingRangeClientCapabilities(!Matched::Union{Missing, Bool}, !Matched::Union{Missing, Int64}, !Matched::Union{Missing, Bool}) at /Users/roiholtzman/.julia/packages/LanguageServer/y1ebo/src/protocol/features.jl:217
  ...
Stacktrace:
 [1] LanguageServer.TextDocumentClientCapabilities(::Dict{String,Any}) at /Users/roiholtzman/.julia/packages/JSONRPC/xbAlZ/src/interface_def.jl:56
 [2] LanguageServer.ClientCapabilities(::Dict{String,Any}) at /Users/roiholtzman/.julia/packages/JSONRPC/xbAlZ/src/interface_def.jl:56
 [3] LanguageServer.InitializeParams(::Dict{String,Any}) at /Users/roiholtzman/.julia/packages/LanguageServer/y1ebo/src/protocol/initialize.jl:111
 [4] dispatch_msg(::JSONRPC.JSONRPCEndpoint{Base.PipeEndpoint,Base.PipeEndpoint}, ::JSONRPC.MsgDispatcher, ::Dict{String,Any}) at /Users/roiholtzman/.julia/packages/JSONRPC/xbAlZ/src/typed.jl:64
 [5] run(::LanguageServerInstance) at /Users/roiholtzman/.julia/packages/LanguageServer/y1ebo/src/languageserverinstance.jl:308
 [6] top-level scope at none:1

Process julia-ls stderr finished
```

in the `lsp-log` there is:
```
Command "julia --startup-file=no --history-file=no -e using InteractiveUtils, Sockets, SymbolServer, LanguageServer; server = LanguageServer.LanguageServerInstance( stdin, stdout,"/Users/roiholtzman/Projects/07_reset_systems/07_code/LocalResetExclusionProcessWatson/","", nothing, "/Users/roiholtzman/.julia/symbolstorev2-lsp-julia"); run(server);" is present on the path.
Command "julia --startup-file=no --history-file=no -e using InteractiveUtils, Sockets, SymbolServer, LanguageServer; server = LanguageServer.LanguageServerInstance( stdin, stdout,"/Users/roiholtzman/Projects/07_reset_systems/07_code/LocalResetExclusionProcessWatson/","", nothing, "/Users/roiholtzman/.julia/symbolstorev2-lsp-julia"); run(server);" is present on the path.
Command "julia --startup-file=no --history-file=no -e using InteractiveUtils, Sockets, SymbolServer, LanguageServer; server = LanguageServer.LanguageServerInstance( stdin, stdout,"/Users/roiholtzman/Projects/07_reset_systems/07_code/LocalResetExclusionProcessWatson/","", nothing, "/Users/roiholtzman/.julia/symbolstorev2-lsp-julia"); run(server);" is present on the path.
Found the following clients for /Users/roiholtzman/Projects/07_reset_systems/07_code/LocalResetExclusionProcessWatson/src/dynamics.jl: (server-id julia-ls, priority 0)
The following clients were selected based on priority: (server-id julia-ls, priority 0)
```

**Comment by juanscr** [comment]
See this [issue](https://github.com/non-Jedi/lsp-julia/issues/35).

TL;DR: Add to your config file `(setq lsp-enable-folding t)` to fix it.

**Comment by roiholtzman** [comment]
> See this [issue](https://github.com/non-Jedi/lsp-julia/issues/35).
> 
> TL;DR: Set `(setq lsp-enable-folding t)` to fix it.

thanks!
I added this to my configuration:
```
(setq lsp-julia-package-dir nil)
(setq lsp-julia-default-environment "~/.julia/environments/v1.5")
(setq lsp-enable-folding t)
```

Now I get the following error:
```
ERROR: MethodError: no method matching Union{Int64, String}(::String)
Stacktrace:
 [1] LanguageServer.InitializeParams(::Dict{String,Any}) at /Users/roiholtzman/.julia/packages/LanguageServer/y1ebo/src/protocol/initialize.jl:111
 [2] dispatch_msg(::JSONRPC.JSONRPCEndpoint{Base.PipeEndpoint,Base.PipeEndpoint}, ::JSONRPC.MsgDispatcher, ::Dict{String,Any}) at /Users/roiholtzman/.julia/packages/JSONRPC/xbAlZ/src/typed.jl:64
 [3] run(::LanguageServerInstance) at /Users/roiholtzman/.julia/packages/LanguageServer/y1ebo/src/languageserverinstance.jl:308
 [4] top-level scope at none:1

Process julia-ls stderr finished
```

**Comment by juanscr** [comment]
Is there a way to replicate this error? I've never seen that, so maybe some steps to replicate it on my machine would be useful.

**Comment by roiholtzman** [comment]
> Is there a way to replicate this error? I've never seen that, so maybe some steps to replicate it on my machine would be useful.

@juanscr  I am not sure. It seems to me that I didn't do anything special, I am using doom emacs, I installed the Julia LanguageServer, and I can't get the lsp to work in emacs.

**Comment by xukai92** [comment]
I have exact the same issue. All I do is enable `(julia +lsp)` and putting the three lines in my `config.el`
```elisp
(setq lsp-julia-package-dir nil)
(setq lsp-julia-default-environment "~/.julia/environments/v1.6")
(setq lsp-enable-folding t)
```

**Comment by xukai92** [comment]
I resolved this issue by upgrading Doom Emacs.
I believe this issue is related to this commit f82a3c692bf4fae029d25948dcb4fb6db6fcc002,
which on its own seems to be a "hack" to make things work in LS.jl?

**Comment by gdkrmr** [comment]
Yes, resolving this took quite a while to get it to work and still feels somewhat hacky.

**Comment by roiholtzman** [comment]
that't great! upgrading also solved it for me.
